### ISSUE TITLE
empty array evals to false

### ISSUE THREAD

**Issue by barisusakli** [benchpressjs/benchpressjs#107]
I think should eval to true like javascript
```
{{{ if emptyArray }}}
doesnt hit there
{{{ end }}}
```

**Comment by julianlam** [comment]
Would be a breaking change then, by the way.

Most usages currently use `{{{ if emptyArray.length }}}` which should work as expected?

**Comment by pitaj** [comment]
This was explicitly chosen to be the case for backwards compatibility. The behavior is documented. Why specifically do you want it changed?

**Comment by barisusakli** [comment]
It is confusing because `if (array)` is always true in js but false in the templates. And `array.length` should be used to check for empty array. Yeah could be major version bump since its a breaking change.

**Comment by pitaj** [comment]
Yes JS does it differently. But the templates are a different language (even if they do compile to JS). Do you have any use cases where this makes a difference?

**Comment by barisusakli** [comment]
I ran into it while working on browsing users https://github.com/barisusakli/nodebb-plugin-browsing-users/blob/master/index.js#L39 I just set it to true now and the template doesnt render anything with {{{ each browsingUsers }}}. Before that it was `= []` but that evaluated to false.

**Comment by barisusakli** [comment]
>One exception is that an empty array is treated as a falsy value, so would not output anything.
This should be on the readme in bold if we are not fixing this 😆
